app/models/item.rb

```diff
@@ -25,4 +25,20 @@ def no_orders?
     item_orders.empty?
   end
 
+  def self.most_popular(num_items_to_show)      
+    Item.joins(:item_orders)
+        .select('items.*, sum(item_orders.quantity)')
+        .group(:id)
+        .order("sum(item_orders.quantity) DESC")
+        .limit(num_items_to_show)
+  end
+
+  def self.least_popular(num_items_to_show)      
+    Item.joins(:item_orders)
+        .select('items.*, sum(item_orders.quantity)')
+        .group(:id)
+        .order("sum(item_orders.quantity) ASC")
+        .limit(num_items_to_show)
+  end
+
```

Comment:
It looks like these two queries are the same except for the ASC and DESC, maybe they could be combined and take an "asc_or_desc" argument or something similar to DRY up the code?